### PR TITLE
[clang] Ensure `--print-runtime-dir` path exists

### PR DESCRIPTION
**Pull request by aganea**
Before this PR, `clang --print-runtime-dir` on Windows used to report a non-existent directory if `LLVM_ENABLE_PER_TARGET_RUNTIME_DIR=OFF`.

We now check if any of the known runtime directories exist before printing any of them on stdout. If none exists, we print `(runtime dir is not present)`.

### CODE REVIEW

**Comment by aganea** [comment]
To give the whole picture, the story is that while building the LLVM package on Windows, I was seeing this:
```
[51/52] Running sanitizer_common testsllvm-lit.py: C:\src\git\llvm-project\llvm\utils\lit\lit\llvm\config.py:57: note: using lit tools: C:\Program Files\Git\usr\bin\
llvm-lit.py: C:\src\git\llvm-project\compiler-rt\test\lit.common.cfg.py:60: warning: Path reported by clang does not exist: "C:\src\git\llvm-project\llvm_package_20.0.0\build64\lib\clang\20\lib\x86_64-pc-windows-msvc". This path was found by running ['C:/src/git/llvm-project/llvm_package_20.0.0/build64/./bin/clang.exe', '--target=x86_64-pc-windows-msvc', '-Wthread-safety', '-Wthread-safety-reference', '-Wthread-safety-beta', '-print-runtime-dir'].
llvm-lit.py: C:\src\git\llvm-project\compiler-rt\test\lit.common.cfg.py:60: warning: Path reported by clang does not exist: "C:\src\git\llvm-project\llvm_package_20.0.0\build64\lib\clang\20\lib\x86_64-pc-windows-msvc". This path was found by running ['C:/src/git/llvm-project/llvm_package_20.0.0/build64/./bin/clang.exe', '--target=x86_64-pc-windows-msvc', '-Wthread-safety', '-Wthread-safety-reference', '-Wthread-safety-beta', '-print-runtime-dir'].
```
This is, we don't set `LLVM_ENABLE_PER_TARGET_RUNTIME_DIR=ON` when building compiler-rt, so we end up taking this cmake codepath: https://github.com/llvm/llvm-project/blob/main/compiler-rt/cmake/base-config-ix.cmake#L106 which ends up with the compiler runtime libs in `{build_folder}\lib\clang\{version}\lib\windows`. However `ToolChain::getRuntimePath()` checks for the triple folder, as in `{build_folder}\lib\clang\{version}\lib\x86_64-pc-windows-msvc` which don't exist. `clang -print-runtime-dir` will just display that, without checking the folder's existence. However if we do `clang-cl a.cpp -fsanitize=address` then we check the library paths (which runtime is part of at this point) for existence: https://github.com/llvm/llvm-project/blob/main/clang/lib/Driver/ToolChains/MSVC.cpp#L153

I think we want to keep the dynamic aspect on supporting both `LLVM_ENABLE_PER_TARGET_RUNTIME_DIR=ON/OFF` at runtime in Clang like @mstorsjo was suggesting in https://discourse.llvm.org/t/runtime-directory-fallback/76860/2, I feel we should check the existence of the path before printing it, thus this PR.

**Comment by aganea** [comment]
Ping.

This still occurs on Windows as of today on main, `clang-cl -print-runtime-dir` doesn't print the right folder path.
On disk we have: `C:\Program Files\LLVM\lib\clang\22\lib\windows`,
whereas `-print-runtime-path` prints the full target as on the other platforms: `C:\Program Files\LLVM\lib\clang\22\lib\x86_64-pc-windows-msvc`.

What is it that we want to do here? Add a dynamic behavior as in this PR, that checks if the path is there, and adjusts accordingly?Change the "windows" component to the target, as the other plateforms (this seems to be some work). Or conversly, accept that on Windows the path is "windows" and not the target? Anything else?

**Review by rnk — APPROVED** [review]
I found this at the bottom of my inbox after a long cleanout.

I'm struggling to remember it at this point in the evening, but we've done things like this before as part of the runtime path changes, and I don't see any reason to object to this one.

Part of me wants to test this behavior since we have the indirection layer, we don't have to copy the clang binary and create an entire runtime tree in the test directory, but I wouldn't block this over that concern.

In conclusion, I think we should merge the patch.
